code/parachain/frame/assets-registry/src/lib.rs

```diff
@@ -309,6 +315,24 @@ pub mod pallet {
 		) -> Option<Self::Balance> {
 			<MinFeeAmounts<T>>::get(parachain_id, remote_asset_id)
 		}
+
+		fn get_foreign_assets_list() -> Vec<Asset<Self::AssetNativeLocation>> {
+			LocalToForeign::<T>::iter()
+				.map(|(local_id, foreign_metadata)| {
+					let decimals = match foreign_metadata.decimals {
+						Some(exponent) => exponent,
+						_ => 12_u32,
+					};
+
+					Asset {
+						name: None,
+						id: local_id.into() as u64,
```

Comment:
.into() followed by `as` is a bit strange, using .into() only should be enough

Reply:
That's what I though but it's not. `local_id` is a `LocalAssetId` and with `.into()` it turns into a `u128`

Reply:
then you should use `.try_into()` to go from `u128` to `u64`, or change the `Asset.id` type to `u128` - `as` will silently truncate the top 64 bits of `u128` when converting from `u64`

Reply:
changed to u128 👌🏼